src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/controllers/web/WorkflowRunController.kt

```diff
@@ -1,37 +1,96 @@
 package org.vaccineimpact.orderlyweb.controllers.web
 
+import com.google.gson.annotations.SerializedName
 import org.vaccineimpact.orderlyweb.ActionContext
+import org.vaccineimpact.orderlyweb.OrderlyServer
+import org.vaccineimpact.orderlyweb.OrderlyServerAPI
+import org.vaccineimpact.orderlyweb.Serializer
 import org.vaccineimpact.orderlyweb.controllers.Controller
+import org.vaccineimpact.orderlyweb.db.AppConfig
 import org.vaccineimpact.orderlyweb.db.repositories.OrderlyWebWorkflowRunRepository
 import org.vaccineimpact.orderlyweb.db.repositories.WorkflowRunRepository
+import org.vaccineimpact.orderlyweb.models.WorkflowRun
+import org.vaccineimpact.orderlyweb.models.WorkflowRunRequest
 import org.vaccineimpact.orderlyweb.models.WorkflowRunSummary
 import org.vaccineimpact.orderlyweb.viewmodels.WorkflowRunViewModel
-import org.vaccineimpact.orderlyweb.models.WorkflowRun
+import java.time.Instant
 
 class WorkflowRunController(
     context: ActionContext,
-    private val workflowRunRepository: WorkflowRunRepository
+    private val workflowRunRepository: WorkflowRunRepository,
+    private val orderlyServerAPI: OrderlyServerAPI
 ) : Controller(context)
 {
-    constructor(context: ActionContext) : this(context, OrderlyWebWorkflowRunRepository())
-
-    fun getWorkflowRunDetails(): WorkflowRun
-    {
-        val key = context.params(":key")
-        return workflowRunRepository.getWorkflowRunDetails(key)
-    }
+    constructor(context: ActionContext) : this(
+        context,
+        OrderlyWebWorkflowRunRepository(),
+        OrderlyServer(AppConfig()).throwOnError()
+    )
 
     @Template("run-workflow-page.ftl")
     fun getRunWorkflow(): WorkflowRunViewModel
     {
         return WorkflowRunViewModel(context)
     }
 
+    fun getWorkflowRunDetails(): WorkflowRun
+    {
+        val key = context.params(":key")
+        return workflowRunRepository.getWorkflowRunDetails(key)
+    }
+
     fun getWorkflowRunSummaries(): List<WorkflowRunSummary>
     {
         return workflowRunRepository.getWorkflowRunSummaries(
             context.queryParams("email"),
             context.queryParams("namePrefix")
         )
     }
+
+    internal data class WorkflowRunResponse(
```

Comment:
Have located this here rather than in `models` because the data classes there all relate to (de)serialisation of JSON for front-end. This class is only used by this class, to validate/deserialise response from orderly.server. I'm open to moving it elsewhere.

Reply:
I don't think that centralising them buys us much, given they're only used by their respective classes? A common naming convention might help though.